js/ui/data_grid/ui.data_grid.export.js

```diff
@@ -825,18 +823,33 @@ dataGridCore.registerModule("export", {
 
                 },
 
-                _correctSearchPanelPosition: function(items) {
-                    items.sort(function(itemA, itemB) {
-                        var result = 0;
+                _correctItemsPosition: function(items, exportItems) {
+                    var columnChooserButtonIndex,
+                        searchPanelIndex,
+                        searchPanel;
 
-                        if(itemA.name === "searchPanel" || (itemA.name === "columnChooserButton" && itemB.name !== "searchPanel")) {
-                            result = 1;
-                        } else if(itemB.name === "searchPanel") {
-                            result = -1;
+                    items.filter(function(item, index) {
+                        if(item.name === "searchPanel") {
+                            searchPanelIndex = index;
+                        }
+                        if(item.name === "columnChooserButton") {
+                            columnChooserButtonIndex = index;
                         }
-
-                        return result;
                     });
+
+                    if(columnChooserButtonIndex > -1) {
+                        searchPanel = items.splice(columnChooserButtonIndex, searchPanelIndex > -1 ? 2 : 1);
```

Comment:
You can extract `searchPanelIndex > -1` (here and at `else` branch) to make the code more readable